packages/vsce/src/utils/resourceUtils.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import {
+  CicsCmciRestClient,
+  getResource,
+  IGetResourceUriOptions,
+  IResourceQueryParams,
+  Utils
+} from "@zowe/cics-for-zowe-sdk";
+import { Session } from "@zowe/imperative";
+import constants from "./constants";
+import { getErrorCode } from "./errorUtils";
+
+export async function runGetResource({ session, resourceName, regionName, cicsPlex, params }: {
+  session: Session,
+  resourceName: string,
+  regionName?: string,
+  cicsPlex?: string,
+  params?: { criteria?: string, parameter?: string; queryParams?: IResourceQueryParams; };
+}) {
+  let count = 0;
+  while (count <= 1) {
```

Comment:
Maybe changing `count` to be a more readable boolean could be nice, although it'll achieve the same result. 

I also wonder if there's a way we can achieve this without a while loop, perhaps moving the functionality inside the while to another method, and calling it from the try block AND the catch (if the 401 is received). Perhaps unecessary change at this point though - depends how strongly others feel about a while loop!